cmake/onnxruntime_providers.cmake

```diff
@@ -128,6 +128,9 @@ endif()
 if(onnxruntime_USE_NNAPI_BUILTIN)
   set(PROVIDERS_NNAPI onnxruntime_providers_nnapi)
 endif()
+if(onnxruntime_USE_SNPE)
```

Comment:
This file is getting bigger. Is it possible to separate the snpe cmake stuff in a separate file and invoke it from here? We can then use this file as the index file for all providers.

Reply:
agreed that we should refactor. but not sure it's appropriate in the scope of this PR as we are aiming to merge this week. we should probably have a PR to split out multiple EP's rather than just one.
i'll add a task to address it.

Reply:
I updated it for snpe ep, it worked. yes, we won't cover other ep's in this PR